modules/rtbhouseBidAdapter.js

```diff
@@ -96,30 +63,257 @@ export const spec = {
-            netRevenue: true,
-            currency: 'USD'
-          };
-          bids.push(bid);
-        }
-      });
+    const responseBody = serverResponse.body;
+    if (!utils.isArray(responseBody)) {
+      return [];
     }
+
+    const bids = [];
+    responseBody.forEach(serverBid => {
+      if (serverBid.price === 0) {
+        return;
+      }
+      if (serverBid.adm.startsWith('{"native')) {
+        bids.push(interpretNativeBid(serverBid));
+      } else {
+        bids.push(interpretBannerBid(serverBid));
+      }
+    });
     return bids;
   }
 };
 
+function buildEndpointUrl(region) {
+  return 'https://' + region + '.' + ENDPOINT_URL;
+}
+
+/**
+ * Produces OpenRTB Imp object from a slot config
+ */
+function mapImpression(slot) {
+  return {
+    id: slot.bidId,
+    banner: mapBanner(slot),
+    native: mapNative(slot),
+    tagid: slot.adUnitCode.toString()
+  };
+}
+
+/**
+ * Produces OpenRTB Banner object
+ */
+function mapBanner(slot) {
+  if (slot.mediaType === 'banner' ||
+    utils.deepAccess(slot, 'mediaTypes.banner') ||
+    (!slot.mediaType && !slot.mediaTypes)) {
+    return {
+      w: slot.sizes[0][0],
+      h: slot.sizes[0][1],
+      format: slot.sizes.map(size => ({
+        w: size[0],
+        h: size[1]
+      }))
+    };
+  }
+}
+
+/**
+ * Produces an OpenRTB Site object
+ */
+function mapSite(validRequest) {
+  const pubId = validRequest && validRequest.length > 0
+    ? validRequest[0].params.publisherId : 'unknown';
+  return {
+    publisher: {
+      id: pubId.toString(),
+    },
+    page: utils.getTopWindowUrl(),
+    name: utils.getOrigin()
+  }
+}
+
+/**
+ * Produces an OpenRTB Native object from a slot config
+ */
+function mapNative(slot) {
+  if (slot.mediaType === 'native' || utils.deepAccess(slot, 'mediaTypes.native')) {
+    return {
+      request: {
+        assets: mapNativeAssets(slot)
+      },
+      ver: '1.1'
+    }
+  }
+}
+
+/**
+ * Produces an OpenRTB Native Ads Assets objects
+ */
+function mapNativeAssets(slot) {
+  const params = slot.nativeParams || utils.deepAccess(slot, 'mediaTypes.native');
+  const assets = [];
+  if (params.title) {
+    assets.push({
+      id: OPENRTB.NATIVE.ASSET_ID.TITLE,
+      required: params.title.required ? 1 : 0,
+      title: {
+        len: params.title.len || 140
+      }
+    })
+  }
+  if (params.image) {
+    assets.push({
+      id: OPENRTB.NATIVE.ASSET_ID.IMAGE,
+      required: params.image.required ? 1 : 0,
+      img: mapNativeImage(params.image, OPENRTB.NATIVE.IMAGE_TYPE.MAIN)
+    })
+  }
+  if (params.icon) {
+    assets.push({
+      id: OPENRTB.NATIVE.ASSET_ID.ICON,
+      required: params.icon.required ? 1 : 0,
+      img: mapNativeImage(params.icon, OPENRTB.NATIVE.IMAGE_TYPE.ICON)
+    })
+  }
+  if (params.sponsoredBy) {
+    assets.push({
+      id: OPENRTB.NATIVE.ASSET_ID.SPONSORED,
+      required: params.sponsoredBy.required ? 1 : 0,
+      data: {
+        type: OPENRTB.NATIVE.DATA_ASSET_TYPE.SPONSORED
```

Comment:
missing length - probably would be good to add text length requirements, if applicable

Reply:
Fixed.